zebra-chain/src/parameters/network/subsidy.rs

```diff
@@ -391,10 +407,36 @@ impl ParameterSubsidy for Network {
             Network::Mainnet => NetworkUpgrade::Canopy
                 .activation_height(self)
                 .expect("canopy activation height should be available"),
-            // TODO: Check what zcashd does here, consider adding a field to `testnet::Parameters` to make this configurable.
-            Network::Testnet(_params) => FIRST_HALVING_TESTNET,
+            Network::Testnet(params) => {
+                if params.is_regtest() {
+                    FIRST_HALVING_REGTEST
+                } else if params.is_default_testnet() {
+                    FIRST_HALVING_TESTNET
+                } else {
+                    height_for_halving_index(1, self)
+                        .expect("first halving height should be available")
```

Comment:
Optional: It would be nice to remove any calls to `is_default_testnet()` outside of tests or code that only runs at startup, here we could compute the first halving height when building `testnet::Parameters`, or even the first 3-4 halving heights so they can be read from memory instead of being computed (which could be hard-coded constants for Mainnet). 

I think this is fine for now and that that can be done in another PR since it only affects Testnet, and is only called twice per semantically verified block and once per call to the `get_block_subsidy` RPC method.